activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -385,6 +385,28 @@ def column_name_for_operation(operation, node) # :nodoc:
         visitor.accept(node, collector).value
       end
 
+      def table_name_options(config = ActiveRecord::Base)
+        {
+          table_name_prefix: config.table_name_prefix,
+          table_name_suffix: config.table_name_suffix
+        }
+      end
+
+      # Finds the correct table name given an Active Record object.
+      # Uses the Active Record object's own table_name, or pre/suffix from the
+      # options passed in.
+      def proper_table_name(name, options = table_name_options)
+        if name.respond_to? :table_name
```

Comment:
Will do, this was just copied out of migration.rb verbatim, I'll make the style changes